### PR TITLE
bump Bitcoin v0.28.1

### PR DESCRIPTION
**Pull request by DanGould**
As far as I can tell these changes are necessary for a passing integration test. Before, it was getting stuck parsing the bip21 from string.

Depends [explicitly](https://github.com/DanGould/rust-payjoin/blob/b1f246a77c2b456ab00cfb50c545d2d55951b6ae/bip78/Cargo.toml#L24) on [bumping bitcoin in the bip21 repo](https://github.com/Kixunil/bip21/pull/6)

### CODE REVIEW

File: bip78/src/fee_rate.rs [hunk 1]
```diff
@@ -46,6 +46,14 @@ impl Mul<FeeRate> for u64 {
     }
 }
 
+impl Div<Weight> for FeeRate {
+    type Output = FeeRate;
```
Comment (Kixunil) [inline bip78/src/fee_rate.rs:50]: The unit doesn't make sense. `FeeRate` = `Amount` / `Weight`, so `FeeRate` / `Weight` should be `Amount` / (`Weight` ^ 2) and there's no such unit in Bitcoin.

File: bip78/src/output_type.rs [hunk 2]
```diff
@@ -1,25 +0,0 @@
...
-        match self {
-            P2Pkh => Weight::from_non_witness_data_size(1 /* OP_DUP */ + 1 /* OP_HASH160 */ + 1 /* OP_PUSH */ + 160 / 8 /* ripemd160 hash size */ + 1 /* OP_EQUALVERIFY */ + 1 /* OP_CHECKSIG */),
-            P2Sh => Weight::from_non_witness_data_size(1 /* OP_HASH160 */ + 1 /* OP_PUSH */ + 160 / 8 /* ripemd160 hash size */ + 1 /* OP_EQUAL */),
-            SegWitV0 { ty: _, nested: true } => Weight::from_non_witness_data_size(1 /* OP_HASH160 */ + 1 /* OP_PUSH */ + 160 / 8 /* ripemd160 hash size */ + 1 /* OP_EQUAL */),
-            SegWitV0 { ty: SegWitV0Type::Pubkey, nested: false } => Weight::from_non_witness_data_size(1 /* OP_PUSH0 */ + 1 /* OP_PUSH */ + 160 / 8 /* ripemd160 hash size */),
-            SegWitV0 { ty: SegWitV0Type::Script, nested: false } => Weight::from_non_witness_data_size(1 /* OP_PUSH0 */ + 1 /* OP_PUSH */ + 256 / 8 /* ripemd160 hash size */),
-            Taproot => Weight::from_non_witness_data_size(1 /* OP_PUSH0 */ + 1 /* OP_PUSH */ + 256 / 8 /* ripemd160 hash size */),
-        }
-    }
-}
```
Comment (Kixunil) [inline bip78/src/output_type.rs:25]: I wanted to provide some way of output type matching but I guess we can remove it for now and revert later.

**Review by Kixunil — CHANGES_REQUESTED** [review]
As I said in the other issue, let's just wait for 0.29.

Also, `Div` is definitely wrong.

**Comment by Kixunil** [comment]
0.29 is out if you'd like to go ahead.

**Comment by DanGould** [comment]
closed by #36

**Pull request closed without merging**